thesslstore_module.php

```diff
@@ -2151,7 +2157,7 @@ private function getProducts() {
         $this->log($this->api_partner_code . "|ssl-products", serialize($product_query_request), "input", true);
         $productsArray = $this->parseResponse($api->product_query($product_query_request));
 
-        return $productsArray;
+        return !empty($productsArray) ? $productsArray : array();
```

Comment:
It's possible the API request may be non-empty, but not an array. I would check both. The default value should be array short syntax, `[]`.

Reply:
Updated

Reply:
I would actually expect `(!empty(...) && is_array(...) ? ... : [])` to make sure it always returns an array like it says it will

Reply:
... Wow, that's a pretty brain dead moment on my part. Updated.